### ISSUE TITLE
Content of cite directive being transformed when it shouldn't be: "..." to ellipsis

### ISSUE THREAD

**Issue by jonathansick** [sphinx-doc/sphinx#3917]
The new smartypants-like approach for transforming text into unicode characters in Sphinx 1.6 is causing us problems.

We have a citation like this:

```rst
:cite:`2016A&A...595A...1G`
```

and it turns out that Sphinx 1.6 is converting these triple periods into a unicode ellipsis. This is breaking the [sphinxcontrib-bibtex](https://github.com/mcmtroffaes/sphinxcontrib-bibtex) resolution of the citation to the BibTeX bibitem:

```
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/timj/work/lsst/dmtn-046/index.rst: WARNING: Citation [2016A&A…595A…1G] is not referenced.
done
preparing documents... done
writing output... [100%] index
/Users/timj/work/lsst/dmtn-046/index.rst:323: WARNING: citation not found: 2016A&A...595A...1G
WARNING: could not relabel citation [2016A&A…595A…1G]
```

(Originally reported in https://github.com/lsst-dm/dmtn-046/pull/1/commits/34d68fd7648c1d56d9e0c8d5e5bf13251a647a7a)

Can we add `cite` to the list of directives that shouldn't be processed by the unicode text converter?

### Environment info
- OS: <Unix/Linux/Mac/Win/other with version>
- Python version: 3.5
- Sphinx version: 1.6.3

**Comment by tk0miya** [comment]
Hmm? The spec of reST says citation labels are built on following rule:

>Simple reference names are single words consisting of alphanumerics plus isolated (no two adjacent) internal hyphens, underscores, periods, colons and plus signs; no whitespace or other characters are allowed. 

So it should not contain `&` mark and consecutive periods.

How did you create such citation definition? By defaullt, sphinx does not recognize the following mark up as citation defs.
```
.. [2016A&A...595A...1G] citation test
```

**Comment by jonathansick** [comment]
The reST spec could be problematic for us. We're working in astronomy and the main source of bibtex citations for us is https://ui.adsabs.harvard.edu. ADS produces bib items with keys that contain `...` and so on. We're trying to stick with the ADS bib keys because they're considered a standard for us.

Example: https://ui.adsabs.harvard.edu/#abs/2016A&A...595A...1G/abstract

```
@ARTICLE{2016A&A...595A...1G,
   author = {{Gaia Collaboration} and {Prusti}, T. and {de Bruijne}, J.~H.~J. and 
	{Brown}, A.~G.~A. and {Vallenari}, A. and {Babusiaux}, C. and 
	{Bailer-Jones}, C.~A.~L. and {Bastian}, U. and {Biermann}, M. and 
	{Evans}, D.~W. and et al.},
    title = "{The Gaia mission}",
  journal = {\aap},
archivePrefix = "arXiv",
   eprint = {1609.04153},
 primaryClass = "astro-ph.IM",
 keywords = {space vehicles: instruments, Galaxy: structure, astrometry, parallaxes, proper motions, telescopes},
     year = 2016,
    month = nov,
   volume = 595,
      eid = {A1},
    pages = {A1},
      doi = {10.1051/0004-6361/201629272},
   adsurl = {http://adsabs.harvard.edu/abs/2016A%26A...595A...1G},
  adsnote = {Provided by the SAO/NASA Astrophysics Data System}
}
```

**Comment by jfbu** [comment]
I think Docutils 0.14 is soon to be released. But it does not yet seem available however one can do

```
pip install docutils==0.14rc2
```

then you can add this to the conf.py of your Sphinx project:

```
def setup(app):
    import sphinx
    from sphinx.transforms import SphinxSmartQuotes
    sphinx.transforms.SphinxSmartQuotes.smartquotes_action = 'qD'
```

Then try make clean html, and the smart quotes should not create ellipses. (the default is `'qDe'` and the `e` is for ellipses.)

For Docutils 0.13.1, Sphinx has to do some patching to fix some other issues. Unfortunately I missed the opportunity to add some user-customizability of the smart quotes action in the patches done in file https://github.com/sphinx-doc/sphinx/blob/stable/sphinx/util/smartypants.py. Those patches are applied only to Docutils < 0.14.

If you are stuck with Docutils 0.13.1, you would need to add to your conf.py a copy of the `educate_tokens` as in this file, modify it by commenting out the line corresponding to https://github.com/sphinx-doc/sphinx/blob/stable/sphinx/util/smartypants.py#L291, and then do a 

```
smartquotes.educate_tokens = my_patched_educate_tokens
```

for your own monkey-patch overwriting the one done in https://github.com/sphinx-doc/sphinx/blob/stable/sphinx/util/smartypants.py by Sphinx itself.

It is much simpler however as indicated above with Docutils 0.14rc2, and perhaps even in future there will be some ``docutils.conf`` variable for user customization of ``smartquotes_action``. As far as I know, there is no such Docutils user variable so far.

But we could have our own config variable for Sphinx users, which would customize the `smartquotes_action` class attribute which ``SphinxSmartQuotes`` inherits from the Docutils `SmartQuotes`.

All the above with Sphinx 1.6.3 or later.

**Comment by jfbu** [comment]
I think your target is LaTeX PDF, but my previous comment apply the same and you should be able to confirm that `make latexpdf` works as expected.

**Comment by tk0miya** [comment]
Fixed by #4504 
Thank you for reporting
